docs: improve doctests for complex number instances in `strided/base/zmap`

prajjwalbajpai opened the pull request:
Resolves a part of #8641.

## Description

This pull request:

-  Updates REPL and documentation examples in the `strided/base/zmap` to use the compact complex instance doctest notation.
-  Replaces example decomposition using realf/imagf with concise doctest annotations.


## Related Issues


This pull request has the following related issues:

-   #8641

## Questions


No.

## Other


No.

## Checklist

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

-   [ ] Yes
-   [x] No

If you answered "yes" above, how did you use AI assistance?

-   [ ] Code generation (e.g., when writing an implementation or fixing a bug)
-   [ ] Test/benchmark generation
-   [ ] Documentation (including examples)
-   [ ] Research and understanding

#### Disclosure

N.A.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

prajjwalbajpai committed:
docs: improve doctests for complex number instances in `strided/base/zmap`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/strided/base/zmap/README.md:
```diff
@@ -52,13 +52,7 @@ var y = new Complex128Array( x.length );
 zmap( x.length, x, 1, y, 1, cceil );
 
 var v = y.get( 0 );
-// returns <Complex128>
```
@prajjwalbajpai You also need to remove any applicable import statements above and below in this file.

prajjwalbajpai replied:
Is there way to rewrite scale function without using `real` and `imag` function or should I leave it as it is?
``` typescript
* @example
* var Complex128Array = require( '@stdlib/array/complex128' );
* var real = require( '@stdlib/complex/float64/real' );
* var imag = require( '@stdlib/complex/float64/imag' );
* var Complex128 = require( '@stdlib/complex/float64/ctor' );
*
* function scale( x ) {
*     var re = real( x );
*     var im = imag( x );
*     return new Complex128( re*10.0, im*10.0 );
* }
```

kgryte replied:
You should leave as is.

kgryte commented on lib/node_modules/@stdlib/strided/base/zmap/docs/types/index.d.ts:
```diff
@@ -64,13 +64,7 @@ interface Routine {
-	*
-	* var im = imag( v );
-	* // returns 10.0
+	* // returns <Complex128>[ 10.0, 10.0 ]
```
@prajjwalbajpai You need to remove any applicable import statements in this and other examples in this file.

kgryte commented on lib/node_modules/@stdlib/strided/base/zmap/lib/index.js:
```diff
@@ -42,13 +42,7 @@
-*
-* var im = imag( v );
-* // returns 10.0
+* // returns <Complex128>[ 10.0, 10.0 ]
```
@prajjwalbajpai You need to remove any applicable import statements in this and other examples in this file.

kgryte commented on lib/node_modules/@stdlib/strided/base/zmap/lib/main.js:
```diff
@@ -55,13 +55,7 @@ var ndarray = require( './ndarray.js' );
-*
-* var im = imag( v );
-* // returns 10.0
+* // returns <Complex128>[ 10.0, 10.0 ]
```
@prajjwalbajpai You need to remove any applicable import statements in this and other examples in this file.

kgryte commented on lib/node_modules/@stdlib/strided/base/zmap/lib/ndarray.js:
```diff
@@ -51,13 +51,7 @@
-*
-* var im = imag( v );
-* // returns 10.0
+* // returns <Complex128>[ 10.0, 10.0 ]
```
@prajjwalbajpai You need to remove any applicable import statements in this and other examples in this file.

kgryte reviewed (changes requested):
Left initial comments.

prajjwalbajpai committed:
remove: unnecessary function imports

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

prajjwalbajpai commented:
@kgryte made the required changes. Sorry for the delay.

kgryte reviewed (approved):
LGTM
